File: app/src/main/java/com/app/belikan/FavoriteFishAdapter.java
package com.app.belikan;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class FavoriteFishAdapter extends RecyclerView.Adapter<FavoriteFishAdapter.FavoriteFishViewHolder> {

    private List<Fish> favoriteFishList;

    public FavoriteFishAdapter(List<Fish> favoriteFishList) {
        this.favoriteFishList = favoriteFishList;
    }

    @NonNull
    @Override
    public FavoriteFishViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_fish_favorite, parent, false);
        return new FavoriteFishViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull FavoriteFishViewHolder holder, int position) {
        Fish fish = favoriteFishList.get(position);
        holder.bind(fish);
    }

    @Override
    public int getItemCount() {
        return favoriteFishList.size();
    }

    public static class FavoriteFishViewHolder extends RecyclerView.ViewHolder {

        private TextView tvFishName;
        private TextView tvPrice;
        private TextView tvRating;
        private ImageView imageViewFish;
        private ImageView imageViewLove;

        public FavoriteFishViewHolder(@NonNull View itemView) {
            super(itemView);
            tvFishName = itemView.findViewById(R.id.tvFishName);
            tvPrice = itemView.findViewById(R.id.tvPrice);
            tvRating = itemView.findViewById(R.id.tvRating);
            imageViewFish = itemView.findViewById(R.id.imageViewFish);
            imageViewLove = itemView.findViewById(R.id.imageViewLove);
        }

        public void bind(Fish fish) {
            tvFishName.setText(fish.getName());
            tvPrice.setText(fish.getPrice());
            tvRating.setText(fish.getRating());
            imageViewFish.setImageResource(fish.getImageResource());
            imageViewLove.setImageResource(fish.isFavorite() ? R.drawable.ic_redlove : R.drawable.ic_outlinedlove);
        }

    }
}

File: app/src/main/java/com/app/belikan/FavoriteFishAdapter.kt
package com.app.belikan

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class FavoriteFishAdapter(private val favoriteFishList: List<Fish>) :
    RecyclerView.Adapter<FavoriteFishAdapter.FavoriteFishViewHolder>() {

    private var listener: OnItemClickListener? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FavoriteFishViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_fish_favorite, parent, false)
        return FavoriteFishViewHolder(view)
    }

    override fun onBindViewHolder(holder: FavoriteFishViewHolder, position: Int) {
        val fish = favoriteFishList[position]
        holder.bind(fish)

        holder.itemView.setOnClickListener {
            val adapterPosition = holder.adapterPosition
            if (adapterPosition != RecyclerView.NO_POSITION) {
                val clickedFish = favoriteFishList[adapterPosition]
                clickedFish.isFavorite = !clickedFish.isFavorite
                notifyItemChanged(adapterPosition)
            }
        }
    }

    override fun getItemCount(): Int {
        return favoriteFishList.size
    }

    inner class FavoriteFishViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val tvFishName: TextView = itemView.findViewById(R.id.tvFishName)
        private val tvPrice: TextView = itemView.findViewById(R.id.tvPrice)
        private val tvRating: TextView = itemView.findViewById(R.id.tvRating)
        private val imageViewFish: ImageView = itemView.findViewById(R.id.imageViewFish)
        private val imageViewLove: ImageView = itemView.findViewById(R.id.imageViewLove)

        fun bind(fish: Fish) {
            tvFishName.text = fish.name
            tvPrice.text = fish.price
            tvRating.text = fish.rating
            imageViewFish.setImageResource(fish.imageResource)
            imageViewLove.setImageResource(if (fish.isFavorite) R.drawable.ic_redlove else R.drawable.ic_outlinedlove)
        }
    }

    fun setOnItemClickListener(listener: OnItemClickListener) {
        this.listener = listener
    }

    interface OnItemClickListener {
        fun onItemClick(position: Int)
    }
}
